File: src/ruby/chap1/coin.rb
(0..10).each do |x|
  (0..3).each do |y|
    (0..10).each do |z|
      cost = 500 * x + 100 * y + 50 * z
      next if cost != 3950
      puts("500円x#{x} + 100円x#{y} + 50円x#{z}=#{cost}")
    end
  end
end

File: src/rust/chap1/coin.rs
fn main() {
    for x in 0..11 {
        for y in 0..4 {
            for z in 0..11 {
                let cost = 500 * x + 100 * y + 50 * z;
                if cost == 3950 {
                    println!("500円x{} + 100円x{} + 50円x{} = {}", x, y, z, cost);
                }
            }
        }
    }
}